### PR TITLE
Ctrl-click on results

### PR DESCRIPTION
**Pull request by s-pace**
PR that:
- Fix the highlighting of lvl1 and lvl2
- Enable command + click
- Enable link preview
- ~Enable multi input~
- Add a playground to use served assets

### CODE REVIEW

**Comment by s-pace** [comment]
Live demo of the new search https://docsearch-pr517.surge.sh/

File: src/lib/templates.js [hunk 1]
```diff
@@ -67,7 +71,9 @@ const templates = {
   `,
   footer: `
     <div class="${footerPrefix}">
-      Search by <a class="${footerPrefix}--logo" href="https://www.algolia.com/docsearch">Algolia</a>
+      <a class="${footerPrefix}--logo" aria-label="Search by algolia" href="https://www.algolia.com/docsearch">
```
Comment (pixelastic) [inline src/lib/templates.js]:
```suggestion
      <a class="${footerPrefix}--logo" aria-labelledby="Search by Algolia" href="https://www.algolia.com/docsearch">
```
Seems like `aria-labelledby` is preferred over `aria-label` when the value of the label is already visible ([source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute)).

I wonder if we also need to add the `title` element?

File: src/lib/templates.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
 const templates = {
   suggestion: `
-  <div class="${suggestionPrefix}
+  <a class="${suggestionPrefix}
     {{#isCategoryHeader}}${suggestionPrefix}__main{{/isCategoryHeader}}
     {{#isSubCategoryHeader}}${suggestionPrefix}__secondary{{/isSubCategoryHeader}}
-  ">
+    "
+    target="_blank" rel="noopener noreferrer" href="{{{url}}}"
```
[the remark this answers is not in the document]
Comment (pixelastic) [reply src/lib/templates.js]: Also, we should add `title` or `aria` attribute to the link

File: src/lib/templates.js [hunk 3]
```diff
@@ -67,7 +71,9 @@ const templates = {
   `,
   footer: `
     <div class="${footerPrefix}">
-      Search by <a class="${footerPrefix}--logo" href="https://www.algolia.com/docsearch">Algolia</a>
+      <a class="${footerPrefix}--logo" aria-label="Search by algolia" href="https://www.algolia.com/docsearch">
+        <img src="${searchByAddress}" alt="Search by algolia" height="100%" width="100%">
```
Comment (pixelastic) [inline src/lib/templates.js]: I don't think we should be using an external url for the image. I would rather have it either as an inline SVG in the JavaScript, or as a background image from the CSS. In any case, if we use an URL, we should use one that targets an origin we control (ie. something in the DocSearch repo, or DocSearch website, not on the main Algolia website).

I'm gonna ping @JonasBa on this: How do you suggest we include the "Powered by Algolia" logo in DocSearch? As an inline SVG, a background image or a hotlink?

File: src/styles/main.scss [hunk 4]
```diff
@@ -2,9 +2,9 @@
 @import "dropdown";
 @import "searchbox";
 
-$searchbox: true!default;
+$searchbox: true !default;
```
Comment (pixelastic) [inline src/styles/main.scss]: I'm a bit rusty on scss syntax, but would those spaces change the meaning or is it ok?

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": true
```
[the remark this answers is not in the document]
Comment (s-pace) [reply .prettierrc]: Yes and because scss is recommending simple one

File: scripts/playground.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/bootstrap/3.3.6/css/bootstrap.min.css" />
+    <!-- at the end of the HEAD -->
+    <link rel="stylesheet" href="http://127.0.0.1:8080/docsearch.min.css" />
+
+</head>
+
+<body>
+    <div class="container">
+        <div class="mt-1">
+            <h1> Search thanks to the assests serves http://127.0.0.1:8080/ </h1>
```
[the remark this answers is not in the document]
Comment (s-pace) [reply scripts/playground.html]: Just to say that it import libs from localhost. It is the best way to use assets serve in localhost

File: scripts/playground.html [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+
+    <!-- at the end of the BODY -->
+    <script type="text/javascript" src="http://127.0.0.1:8080/docsearch.min.js"></script>
+    <script type="text/javascript"> docsearch({
+            apiKey: '25626fae796133dc1e734c6bcaaeac3c',
+            indexName: 'docsearch',
+            inputSelector: ['#q', '#z'],
```
[the remark this answers is not in the document]
Comment (s-pace) [reply scripts/playground.html]: Good idea

File: scripts/serve [hunk 8]
```diff
@@ -3,7 +3,9 @@
 yarn run build:css
 
 echo "Find all files at http://127.0.0.1:8080/";
+open scripts/playground.html
```
[the remark this answers is not in the document]
Comment (s-pace) [reply scripts/serve]: Since it is really fast. I will not bother. If you have a quick solution, feel free to LMK

File: src/lib/DocSearch.js [hunk 9]
```diff
@@ -64,7 +64,7 @@ class DocSearch {
     this.apiKey = apiKey;
     this.appId = appId;
     this.indexName = indexName;
-    this.input = DocSearch.getInputFromSelector(inputSelector);
+    this.input = DocSearch.getInputsFromSelector(inputSelector);
```
[the remark this answers is not in the document]
Comment (s-pace) [reply src/lib/DocSearch.js]: Ok I will enhance it with the string selectors

File: src/lib/DocSearch.js [hunk 10]
```diff
@@ -79,38 +79,44 @@ class DocSearch {
...
+          {
+            source: this.getAutocompleteSource(transformData, queryHook),
+            templates: {
+              suggestion: DocSearch.getSuggestionTemplate(this.isSimpleLayout),
+              footer: templates.footer,
+              empty: DocSearch.getEmptyTemplate(),
+            },
+          },
+        ]);
+        if (handleSelected) {
```
[the remark this answers is not in the document]
Comment (s-pace) [reply src/lib/DocSearch.js]: Yes this is why this one is only here to ensure backward compatibility. It enables method that were overridden

File: src/lib/templates.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply src/lib/templates.js]: Not needed in fact. I will remove it and add the `title` & `aria`

Are you fine with an image within a a tag?

File: src/lib/templates.js (hunk 3 above)
Comment (s-pace) [reply src/lib/templates.js]: I will use https://instantsearch-css.netlify.com/widgets/powered-by/ instead

File: src/styles/main.scss (hunk 4 above)
Comment (s-pace) [reply src/styles/main.scss]: It is ok

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (pixelastic) [reply .prettierrc]: Ah great. In that case you can apply the `singleQuote` only to SCSS files like this:

```json
{
   "overrides": [
    {
      "files": "*.scss",
      "options": {
        "singleQuote": true
      }
    }
  ]
}
```

This way it won't change the way to prettify js

File: scripts/playground.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (pixelastic) [reply scripts/playground.html]: Ah ok, so I would reword as:
```suggestion
            <h1> Search thanks to the assets served at http://127.0.0.1:8080/ </h1>
```

File: scripts/serve (hunk 8 above)
[the remark this answers is not in the document]
Comment (pixelastic) [reply scripts/serve]: Ok, let's keep it that way, we'll fix if we see it's an issue

File: src/lib/DocSearch.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (pixelastic) [reply src/lib/DocSearch.js]: Did you double check that it was still working? Because we now have links instead of divs, it's possible the default `click` event from the browser (moving to the next page) is triggered before the custom `handleSelected` and thus `handleSelected` will never be triggered.

If `handleSelected` is custom-defined, we should make sure the default event on the link is captured and redirected to `handleSelected`

File: src/lib/templates.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pixelastic) [reply src/lib/templates.js]: I started a discussion about that below :)

File: src/lib/templates.js (hunk 3 above)
Comment (pixelastic) [reply src/lib/templates.js]: Oh neat.

File: src/styles/main.scss (hunk 4 above)
Comment (pixelastic) [reply src/styles/main.scss]: :+1:

File: src/lib/DocSearch.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply src/lib/DocSearch.js]: No I haven't run any test so far except human emulation

File: .prettierrc (hunk 5 above)
[the remark this answers is not in the document]
Comment (s-pace) [reply .prettierrc]: Solved

**Comment by s-pace** [comment]
@pixelastic I wanted to resolve automatically the conflicts. I have used the hint provided by github. It actually merges everything.
Feel free to open a PR to add test, then we can deploy it

**Comment by pixelastic** [comment]
Wait, we should not release this as v3, but keep it a minor bump.

If we release it as v3, people won't be able to use it right away, as it would require them to update the link to the CDN:
image

We need to make sure it's still a v2.x, with backward compatibility. We want to keep the big bump to v3 for the rewrite, where we'll be able to break backward compat if needed.

**Comment by s-pace** [comment]
I think we do not have any issue with releasing it straight away. I do not see any breaking change. Use os really regular so let's release the v2 :)
(We will help people to solve issue if needed)

**Comment by s-pace** [comment]
Could you have a look regarding gzip or any way to reduce the size of it pls?

**Comment by pixelastic** [comment]
> I think we do not have any issue with releasing it straight away. 
We're still missing at least one test. I would not release something that hasn't been tested

> Could you have a look regarding gzip or any way to reduce the size of it pls?
Yes, and I think the logo display is bigger in that branch than it used to be. I'd like to double check that before the release as well.

**Comment by s-pace** [comment]
Fine with it. Write a test and then we can release 🎉
